File: internal/tui/commands.go
package tui

import "strings"

type commandInfo struct {
	Name  string
	Usage string
	Desc  string
}

func (c commandInfo) display() string {
	if c.Usage != "" {
		return c.Usage
	}
	return c.Name
}

var commands = []commandInfo{
	{Name: "/done", Desc: "review the step"},
	{Name: "/why", Desc: "zoom out"},
	{Name: "/stuck", Desc: "get help"},
	{Name: "/skip", Desc: "next step"},
	{Name: "/recap", Desc: "session recap"},
	{Name: "/run", Usage: "/run [cmd]", Desc: "run code"},
	{Name: "/summary", Desc: "session summary"},
	{Name: "/copy", Desc: "copy session to clipboard"},
	{Name: "/dial", Usage: "/dial <0-3>", Desc: "typing dial"},
	{Name: "/profile", Desc: "adjust profile"},
	{Name: "/help", Desc: "list commands"},
	{Name: "/quit", Desc: "exit"},
}

func matchCommands(text string) []commandInfo {
	if text == "/" {
		return commands
	}
	var out []commandInfo
	for _, c := range commands {
		if strings.HasPrefix(c.Name, text) {
			out = append(out, c)
		}
	}
	return out
}

File: src/nina/tui/commands.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class CommandInfo:
    name: str
    desc: str
    usage: str = ""

    def display(self) -> str:
        return self.usage or self.name


COMMANDS: list[CommandInfo] = [
    CommandInfo("/done", "review the step"),
    CommandInfo("/why", "zoom out"),
    CommandInfo("/stuck", "get help"),
    CommandInfo("/skip", "next step"),
    CommandInfo("/recap", "session recap"),
    CommandInfo("/run", "run code", usage="/run [cmd]"),
    CommandInfo("/summary", "session summary"),
    CommandInfo("/copy", "copy session to clipboard"),
    CommandInfo("/dial", "typing dial", usage="/dial <0-3>"),
    CommandInfo("/auto", "auto-approve commands", usage="/auto <on|off>"),
    CommandInfo("/profile", "adjust profile"),
    CommandInfo("/help", "list commands"),
    CommandInfo("/quit", "exit"),
]


def match_commands(text: str) -> list[CommandInfo]:
    if text == "/":
        return list(COMMANDS)
    return [c for c in COMMANDS if c.name.startswith(text)]
